packages/dds/tree/src/feature-libraries/sequence-field/moveEffectTable.ts

```diff
@@ -581,14 +400,32 @@ export function splitMarkOnInput<TMark extends InputSpanningMark<unknown>>(
         case "ReturnFrom": {
             // TODO: Handle detach index for ReturnFrom
             const newId = genId();
-            splitMoveDest(moveEffects, mark.id, [
-                { id: mark.id, count: length },
-                { id: newId, count: remainder },
-            ]);
-            return [
-                { ...markObj, count: length },
-                { ...markObj, id: newId, count: remainder },
-            ] as [TMark, TMark];
+            splitMove(
+                moveEffects,
+                MoveEnd.Dest,
+                mark.revision ?? revision,
+                mark.id,
+                newId,
+                length,
+                remainder,
+            );
+            if (recordMoveEffect) {
```

Comment:
I don't understand the semantic of the `recordMoveEffect` parameter. It looks like it's being used to control whether the source should also be split. Why do we call `splitMove` for the `MoveEnd.Dest` even when `recordMoveEffect` is false?